### PR TITLE
Converted audio play functions to *_user

### PR DESCRIPTION
**Pull request by IBNobody**
void startup_song_user() {}
void goodbye_song_user() {}
void music_on_user() {}
void audio_on_user() {}
void music_scale_user() {}

I updated the default planck layout to use these.

Also, I fixed the make all target to support parallel building.

### CODE REVIEW

File: tmk_core/common/command.c [hunk 1]
```diff
@@ -358,7 +358,7 @@ static bool command_common(uint8_t code)
             print("\n\nJumping to bootloader... ");
             #ifdef AUDIO_ENABLE
 	            stop_all_notes();
-                play_goodbye_tone();
+                goodbye_song_user();
```
Comment (jackhumbert) [inline tmk_core/common/command.c]: I'm not entirely sure this should be a "song" - what if we just called it `goodbye_user` here, to let people flash leds, play notes, or send keystrokes?

File: quantum/audio/audio.h [hunk 2]
```diff
@@ -87,9 +87,10 @@ void play_notes(float (*np)[][2], uint16_t n_count, bool n_repeat, float n_rest)
 
 bool is_playing_notes(void);
 
-void play_goodbye_tone(void);
-void play_startup_tone(void);
+void startup_song_user(void);
```
Comment (jackhumbert) [inline quantum/audio/audio.h]: Do we really need a dedicated song function with the `matrix_init_user`? We could call it `keyboard_init_user` and move it outside of `matrix.c` and into `/tmk_core/common/keyboard.c`. Maybe `quantum_init`, `user_init`? `keyboard_init` is already taken..

Comment (IBNobody) [inline quantum/audio/audio.h]: I have no preference other than liking start or startup over init.

File: tmk_core/common/command.c (hunk 1 above)
Comment (IBNobody) [inline tmk_core/common/command.c]: startup_user / shutdown_user?

Comment (jackhumbert) [inline tmk_core/common/command.c]: Perfect :)

**Comment by jackhumbert** [comment]
Moved to #349

**Comment by jackhumbert** [comment]
The new ALL outputs all of the commands when running them, which is kind of odd/annoying. It also feels like it might take a little break in-between each, but I might just be imagining that.

**Comment by IBNobody** [comment]
Which ALL do you mean?

**Comment by jackhumbert** [comment]
The `make all` where each process is preceded by `$(MAKE)`.

**Comment by IBNobody** [comment]
Ah, ok. What do you mean by it outputting everything? Are you not running
it with the silent flag?

**Comment by jackhumbert** [comment]
It outputs `/Applications/Xcode.app/Contents/Developer/usr/bin/make end` and etc. for each command - possibly because each has a newline? FTR, I'm always just running `make [keymap=*] [dfu]` :) you run custom arguments, right?

**Comment by IBNobody** [comment]
Without the -s flag, I would always get the full compiler command line,
which was extremely annoying. I started using -s to silence this. I will
remove the -s flag and try running it with my restructured make. Maybe I'm
silencing something you are seeing?

I also run -r because that disables some rule assumptions that aren't
needed when you have a proper make file like we do. Maybe this is a Windows
thing, but not including this would make Make take forever.

**Comment by jackhumbert** [comment]
`-r` doesn't seem to change much for me, but `-s` does remove the command line `../make end` stuff, and looks a lot cleaner in general :) it looks like there are [some options](http://stackoverflow.com/questions/24005166/gnu-make-silent-by-default) for adding it to the Makefile - I wouldn't mind doing that since all errors/warnings/useful info is still outputted.

**Comment by IBNobody** [comment]
Yes yes yes. .SILENT would be a great quality of life improvement for those
who aren't make savvy.

**Pull request closed without merging**